enhancements/kube-controller-manager/max-namespace-size.md

```diff
@@ -0,0 +1,112 @@
+---
+title: limit-resources-per-namespace
+authors:
+  - "@deads2k"
+reviewers:
+  - "@soltysh"
+approvers:
+  - "@mfojtik"
+creation-date: 2020-01-31
+last-updated: 2020-01-31
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Limit Resources Per Namespace
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Create a new `namespace-size-policy-controller` in the existing [cluster-policy-controller](https://github.com/openshift/cluster-policy-controller/).
+This controller will create quota objects in every namespace that limit the size of certain error-prone resources.
+We will start with secrets and configmaps.
+It will be created with a goal of protecting the cluster from accidental runaway resource creation.
+
+## Motivation
+
+Kuberentes clusters are vulnerable to runaway namespace conditions.
+This is true regardless of how the cluster is hosted.
+While the true concern is total size of objects in etcd, the immediate cause is usually an excessive number of resources in a single namespace.
+We can easily carve off enforcement of total object count per namespace in a non-disruptive way that protects against common cases of errors.
+
+### Goals
+
+1. protect the cluster from accidental runaway resource creation.
+2. prevent runaway growth instead of simply warning about large sizes
+
+### Non-Goals
+
+1. thwarting evil users trying to break the cluster
+
+## Proposal
```

Comment:
Should we also Include some restrictions or limit to the number of namespaces?

It seems like we're only addressing half of the DOS situation if we don't also limit the number of namespaces to something like 5000, and provide a way to override or later this after a cluster is installed.